specs/core/0_beacon-chain.md

```diff
@@ -1134,32 +1135,32 @@ Before genesis has been triggered and whenever the deposit contract emits a `Dep
 * `deposits` is the list of all deposits, ordered chronologically, up to and including the deposit triggering the latest `Deposit` log
 * `timestamp` is the Unix timestamp in the Ethereum 1.0 block that emitted the latest `Deposit` log
 
-When `is_genesis_trigger(deposits, timestamp) is True` for the first time let:
+When `is_genesis_trigger(deposits, timestamp, deposit_root) is True` for the first time let:
 
 * `genesis_deposits = deposits`
 * `genesis_time = timestamp - timestamp % SECONDS_PER_DAY + 2 * SECONDS_PER_DAY` where `SECONDS_PER_DAY = 86400`
+* `deposit_root` is the Merkle tree root of the data of the given `deposits`
 * `genesis_eth1_data` be the object of type `Eth1Data` where:
-    * `genesis_eth1_data.block_hash` is the Ethereum 1.0 block hash that emitted the log for the last deposit in `deposits`
     * `genesis_eth1_data.deposit_root` is the deposit root for the last deposit in `deposits`
     * `genesis_eth1_data.deposit_count = len(genesis_deposits)`
+    * `genesis_eth1_data.block_hash` is the Ethereum 1.0 block hash that emitted the log for the last deposit in `deposits`
 
 *Note*: The function `is_genesis_trigger` has yet to be agreed by the community, and can be updated as necessary. We define the following testing placeholder:
 
 ```python
-def is_genesis_trigger(deposits: List[Deposit], timestamp: uint64) -> bool:
+def is_genesis_trigger(deposits: List[Deposit], timestamp: uint64, deposit_root: Hash) -> bool:
     # Process deposits
     state = BeaconState()
-    for deposit in deposits:
-        process_deposit(state, deposit)
+    for index, deposit in enumerate(deposits):
+        process_deposit(state, deposit, deposit_index=index, deposit_root=deposit_root)
 
     # Count active validators at genesis
     active_validator_count = 0
-    for validator in state.validator_registry:
-        if validator.effective_balance == MAX_EFFECTIVE_BALANCE:
+    for validator in state.validators:
+        if validator.effective_balance >= MAX_EFFECTIVE_BALANCE:
```

Comment:
`validator.effective_balance > MAX_EFFECTIVE_BALANCE` is not possible. Seems semantically cleaner to write `validator.effective_balance == MAX_EFFECTIVE_BALANCE`

Reply:
Oh but there's also:
```python
# Process genesis activations
    for validator in state.validators:
        if validator.effective_balance >= MAX_EFFECTIVE_BALANCE:
            validator.activation_eligibility_epoch = GENESIS_EPOCH
            validator.activation_epoch = GENESIS_EPOCH
```
in `get_genesis_beacon_state`. Should they both be `==`?

Reply:
Aha, I see. They are equivalent and I suggest `validator.effective_balance == MAX_EFFECTIVE_BALANCE` in both cases :)